packages/core/threads/thingsboard_thread.py

```diff
@@ -0,0 +1,104 @@
+            if not ThingsBoardThread.should_be_running(config):
+                logger.info("ThingsboardThread shall not be running anymore.")
+                client.disconnect()
+                return
+
+            # publish if client is connected
+            if not client.is_connected():
+                logger.warning(
+                    "Client is currently not connected. Waiting for reconnect."
+                )
+            else:
+                # read latest state file
+                current_state = interfaces.StateInterface.load_state()
+                state: Dict[str, Optional[Union[str, bool, int, float]]] = {
+                    "last_updated": str(current_state.last_updated),
+                    "helios_indicates_good_conditions": current_state.helios_indicates_good_conditions,
+                    "measurements_should_be_running": current_state.measurements_should_be_running,
+                    "memory_usage": current_state.operating_system_state.memory_usage,
+                    "filled_disk_space_fraction": current_state.operating_system_state.filled_disk_space_fraction,
+                    "fan_speed": current_state.plc_state.actors.fan_speed,
+                    "current_angle": current_state.plc_state.actors.current_angle,
+                    "auto_temp_mode": current_state.plc_state.control.auto_temp_mode,
+                    "manual_control": current_state.plc_state.control.manual_control,
+                    "manual_temp_mode": current_state.plc_state.control.manual_temp_mode,
+                    "sync_to_tracker": current_state.plc_state.control.sync_to_tracker,
+                    "humidity": current_state.plc_state.sensors.humidity,
+                    "temperature": current_state.plc_state.sensors.temperature,
+                    "cover_closed": current_state.plc_state.state.cover_closed,
+                    "motor_failed": current_state.plc_state.state.motor_failed,
+                    "rain": current_state.plc_state.state.rain,
+                    "reset_needed": current_state.plc_state.state.reset_needed,
+                    "ups_alert": current_state.plc_state.state.ups_alert,
+                    "heater": current_state.plc_state.power.heater,
+                    "spectrometer": current_state.plc_state.power.spectrometer,
+                }
+
+                telemetry_with_ts = {
+                    "ts": int(round(time.time() * 1000)),
+                    "values": state,
+                }
+
+                try:
+                    # Sending telemetry without checking the delivery status (QoS1)
+                    result = client.send_telemetry(telemetry_with_ts)
+                    logger.info(f"Published with result: {result}")
+                except Exception as e:
+                    logger.exception(e)
+                    logger.info("Failed to publish last telemetry data.")
+
+            time.sleep(config.general.seconds_per_core_interval)
```

Comment:
There should be a separate parameter for the "seconds per thingsboard communication interval".